Fix:

update.vala:191.13-191.52: error: 1 extra arguments for `Archive.Result Archive.Read.open_memory (uint8[])'
update.vala:201.33-201.73: error: 1 extra arguments for `ssize_t Archive.Read.read_data (uint8[])'

https://gitlab.gnome.org/GNOME/libhttpseverywhere/-/issues/1
https://gitlab.gnome.org/GNOME/libhttpseverywhere/-/commit/6da08ef1ade9ea267cecf14dd5cb2c3e6e5e50cb
--- src/update.vala.orig
+++ src/update.vala
@@ -188,7 +188,12 @@ namespace HTTPSEverywhere {
 
             Archive.Read zipreader = new Archive.Read();
             zipreader.set_format(Archive.Format.ZIP);
+#if VALA_0_42
+            output.length = (int) size_read;
+            zipreader.open_memory(output);
+#else
             zipreader.open_memory(output, size_read);
+#endif
 
             string json = "";
             unowned Archive.Entry e = null;
@@ -198,11 +203,15 @@ namespace HTTPSEverywhere {
                     found_ruleset_file = true;
                     uint8[] jsonblock = new uint8[1024*1024];
                     while (true) {
-                        var r = zipreader.read_data(jsonblock, 1024*1024);
+#if VALA_0_42
+                        var r = zipreader.read_data(jsonblock);
+#else
+                        var r = zipreader.read_data(jsonblock, jsonblock.length);
+#endif
                         if (r < 0) {
                             throw new UpdateError.CANT_READ_FROM_ARCHIVE("Failed reading archive stream");
                         }
-                        if (r < 1024*1024 && r != 0) {
+                        if (r < jsonblock.length && r != 0) {
                             uint8[] remainder = new uint8[r];
                             Memory.copy(remainder, jsonblock, r);
                             json += (string)remainder;
